juju metadata generate-image crash on 3.6

### Description

It seems like a regression. Running `juju metadata generate-image` even without any argument causes crash

```
mastier@drakkar:~$ juju metadata generate-image
panic: empty value for "controller-uuid" found in configuration (type <nil>, val <nil>)

goroutine 1 [running]:
github.com/juju/juju/controller.Config.mustString(0xc0007264e0, {0x6069862, 0xf})
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/controller/config.go:647 +0x125
github.com/juju/juju/controller.Config.ControllerUUID(0xc00033a8f8?)
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/controller/config.go:811 +0x1f
main.prepare(0x0?, {0xc0001833e0?, 0xc000131888?}, {0x69160d8?, 0xc00008c6d0?})
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/cmd/plugins/juju-metadata/imagemetadata.go:61 +0x1cf
main.(*imageMetadataCommand).setParams(0xc00066d8c0, 0xc0000fd020, {{0x6043a3e, 0x6}, {{0x6041ee5, 0x5}, {0x6043a44, 0x6}}})
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/cmd/plugins/juju-metadata/imagemetadata.go:142 +0x17e
main.(*imageMetadataCommand).Run(0xc00066d8c0, 0xc0000fd020)
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/cmd/plugins/juju-metadata/imagemetadata.go:246 +0x3de
github.com/juju/juju/cmd/modelcmd.(*controllerCommandWrapper).Run(0xc0006cf770, 0xc0000fd020)
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/cmd/modelcmd/controller.go:403 +0xf1
github.com/juju/juju/cmd/modelcmd.(*baseCommandWrapper).Run(0xc00034def0, 0xc0000fd020)
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/cmd/modelcmd/base.go:569 +0x95
github.com/juju/cmd/v3.(*SuperCommand).Run(0xc00066d760, 0xc0000fd020)
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/vendor/github.com/juju/cmd/v3/supercommand.go:557 +0x369
github.com/juju/cmd/v3.Main({0x68f5cf0, 0xc00066d760}, 0xc0000fd020, {0xc000070090, 0x1, 0x1})
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/vendor/github.com/juju/cmd/v3/cmd.go:468 +0x232
main.Main({0xc000070080, 0x2, 0x2})
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/cmd/plugins/juju-metadata/metadata.go:38 +0xca
main.main()
	/build/snapcraft-juju-d1a21e3c651528a33e682b3dd622daad/parts/juju/build/cmd/plugins/juju-metadata/metadata.go:67 +0x28
```

When I look at the controller-config I see controller-uuid
```
$ juju controller-config controller-uuid
3d7b54d2-554b-428a-8e9c-a746801cb696
```



### Juju version

3.6.9

### Cloud

_No response_

### Expected behaviour

For instance when you just switch back juju client to 3.5 it works, it errors on missing data but no crash

```
mastier@drakkar:~$ sudo snap refresh --channel 3.5/stable juju
juju (3.5/stable) 3.5.7 from Canonical✓ refreshed

mastier@drakkar:~$ juju metadata generate-image
ERROR model "controller" cannot provide region and endpoint
```

### Reproduce / Test

1.  Bootstrap juju 3.6 on localhost
```
snap install --channel=3.6/stable juju
```
current version: installed:       3.6.9                        (32163) 105MB -
```
juju bootstrap localhost
```
2. Wait until the controller is ready
```
juju controller --refresh
Controller            Model      User   Access     Cloud/Region         Models  Nodes    HA  Version
localhost-localhost*  landscape  admin  superuser  localhost/localhost       1      1  none  3.6.9  
```
3. Run command `juju metadata generate-image`

### Notes & References

_No response_

--- a/cmd/plugins/juju-metadata/imagemetadata.go
+++ b/cmd/plugins/juju-metadata/imagemetadata.go
@@ -55,10 +55,14 @@ func prepare(ctx *cmd.Context, controllerName string, store jujuclient.ClientSto
 	// identify region and endpoint info that we need. Not sure what
 	// we'll do about simplestreams.MetadataValidator yet. Probably
 	// move it to the EnvironProvider interface.
+	ctrl, err := store.ControllerByName(controllerName)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
 	return environs.New(context.TODO(), environs.OpenParams{
 		Cloud:          params.Cloud,
 		Config:         cfg,
-		ControllerUUID: bootstrapConfig.ControllerConfig.ControllerUUID(),
+		ControllerUUID: ctrl.ControllerUUID,
 	})
 }
 

--- a/cmd/plugins/juju-metadata/imagemetadata_test.go
+++ b/cmd/plugins/juju-metadata/imagemetadata_test.go
@@ -16,6 +16,7 @@ import (
 	gc "gopkg.in/check.v1"
 
 	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/controller"
 	"github.com/juju/juju/environs/config"
 	"github.com/juju/juju/internal/provider/dummy"
 	"github.com/juju/juju/jujuclient"
@@ -148,8 +149,13 @@ func (s *ImageMetadataSuite) TestImageMetadataFilesLatestLTS(c *gc.C) {
 		"region":          "us-east-1",
 	})
 	c.Assert(err, jc.ErrorIsNil)
+	// The controller config used for bootstrap config
+	// does not have the controller uuid or ca cart.
+	controllerCfg := testing.FakeControllerConfig()
+	delete(controllerCfg, controller.ControllerUUIDKey)
+	delete(controllerCfg, controller.CACertKey)
 	s.store.BootstrapConfig["ec2-controller"] = jujuclient.BootstrapConfig{
-		ControllerConfig: testing.FakeControllerConfig(),
+		ControllerConfig: controllerCfg,
 		Cloud:            "ec2",
 		CloudRegion:      "us-east-1",
 		Config:           ec2Config.AllAttrs(),

--- a/cmd/plugins/juju-metadata/validateimagemetadata_test.go
+++ b/cmd/plugins/juju-metadata/validateimagemetadata_test.go
@@ -14,6 +14,7 @@ import (
 	gc "gopkg.in/check.v1"
 
 	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/controller"
 	corebase "github.com/juju/juju/core/base"
 	"github.com/juju/juju/core/model"
 	"github.com/juju/juju/environs/config"
@@ -134,8 +135,13 @@ func cacheTestEnvConfig(c *gc.C, store *jujuclient.MemStore) {
 		User: "admin",
 	}
 
+	// The controller config used for bootstrap config
+	// does not have the controller uuid or ca cart.
+	controllerCfg := coretesting.FakeControllerConfig()
+	delete(controllerCfg, controller.ControllerUUIDKey)
+	delete(controllerCfg, controller.CACertKey)
 	store.BootstrapConfig["ec2-controller"] = jujuclient.BootstrapConfig{
-		ControllerConfig:    coretesting.FakeControllerConfig(),
+		ControllerConfig:    controllerCfg,
 		ControllerModelUUID: ec2UUID,
 		Config:              ec2Config.AllAttrs(),
 		Cloud:               "ec2",

